src/service/extension-script.js

```diff
@@ -186,28 +189,34 @@ export function getExtensionScripts(win, extensionId, includeInserted = true) {
 
 /**
  * Get list of all the extension JS files.
- * @param {HTMLHeadElement|Element|ShadowRoot} head
- * @return {!Array<string>}
+ * @param {HTMLHeadElement|Element|ShadowRoot|Document} head
+ * @return {!Array<{extensionId: string, extensionVersion: string}>}
  */
 export function extensionScriptsInNode(head) {
   // ampdoc.getHeadNode() can return null.
   if (!head) {
     return [];
   }
-  const scripts = {};
   // Note: Some extensions don't have [custom-element] or [custom-template]
   // e.g. amp-viewer-integration.
   const list = head.querySelectorAll(
     'script[custom-element],script[custom-template]'
   );
+  const scripts = [];
   for (let i = 0; i < list.length; i++) {
     const script = list[i];
-    const name =
+    if (script.hasAttribute('i-amphtml-inserted')) {
```

Comment:
Won's prevent a reloaded script from specifying the correct version? Did you mean `i-amphtml-loaded-new-version `?

Reply:
The main use of the `extensionScriptsInNode` is currently `stubElementsForDoc` which prestubs elements in a doc based on the extensions in the HEAD. This happens when the runtime wakes up for the first time and tries to register stubs. I wanted to avoid it accidentally register stubs for extensions inserted programmatically. For instance, an extension could be inserted programmatically for a FIE embed and shouldn't get automatically stubbed in the parent doc.

For the case with the reloaded extensions - it doesn't matter for this use case. For stubbing it doesn't matter whether an extension is outdated - it only creates a stub and waits for the actual extension to arrive.

If you'd prefer I can make this check optional via an argument. E.g. allowInserted: boolean.

Reply:
> I wanted to avoid it accidentally register stubs for extensions inserted programmatically. For instance, an extension could be inserted programmatically for a FIE embed and shouldn't get automatically stubbed in the parent doc.

Isn't this searching and stubbing inside the FIE doc? I don't think I'm properly understanding.

At the moment, I think leaving this in here, even if behind a boolean, is actually a bug.

Reply:
No, FIE doesn't have any scripts in it, so this kind of stubbing would not work there. The FIE creator is responsible for pre-stubbing it.

Reply:
Actually, I think I agree with you. We never had this check before, so no reason to introduce it here.